src/meshapi/views/forms.py

```diff
@@ -516,6 +517,18 @@ def network_number_assignment(request: Request) -> Response:
 
     # First, we try to identify a Node object for this install if it doesn't already have one
     if not nn_install.node:
+        # Don't allow old recycled or closed installs to have a network number assigned to them,
+        # since this would be very confusing. Installs in this status should re-submit the join
+        # form and try again with a new install number
+        if nn_install.status in [Install.InstallStatus.CLOSED, Install.InstallStatus.NN_REASSIGNED]:
```

Comment:
I guess this is just how it has to be for reassigned NNs, but what do we think about deduping closed installs? Maybe this is an edge case not worth thinking about, but if it takes someone 3 tries to get installed after filling out the join form, they're gonna have three install numbers which would clutter the UI.

Then again, we can just... re-open the request or something.

Reply:
I'm not sure I follow what you're saying about closed installs. Can you elaborate on the case you're worried about?

Reply:
What happens if someone submits the join form, then their request gets closed, then they submit the join form again? Does it re-open the old request or make a new one with the same information?

Reply:
If the request is closed, they will get a new install number

Reply:
I think this is a data safety issue. Unless we can prove that in all cases it would be safe to re-open a closed request, we have to error on the side of caution and issue a new install number